File: 0169-majority-element/0169-majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int count =0;
        int value =0;
        for(int i =0;i<nums.size();i++){
            if(count==0){
                value = nums[i];
            }
            if(value == nums[i]){
                count++;
            }else {
                count--;
            }
        }
        return value;

    }
};

File: 0169-majority-element/0169-majority-element.java
class Solution {
    public int majorityElement(int[] nums) {
        HashMap<Integer, Integer> freq = new HashMap<>();
        for(int num : nums){
            freq.put(num, freq.getOrDefault(num,0)+1);
            if(freq.getOrDefault(num,0) > nums.length/2){
                return num;
            }
        }
        return -1;
    }
}